docs/internal/github_actions.rst

```diff
@@ -0,0 +1,203 @@
+********
+
+GitHub Actions is the continuous integration solution currently being
+used to evaluate the readiness of pull requests. It builds OpenDDS and runs the
+test suite across a wide variety of operation systems and build configurations.
+
+*****************
+Table of Contents
+*****************
+
+* [Legend for GitHub Actions Build Names]
+  * [Operating System]
+  * [Build Configuration]
+  * [Build Type]
+  * [Build Options]
+  * [Feature Mask]
+* [Introduction to GitHub Actions Workflows]
+* [Caching]
+* [Job Types]
+* [Test Results]
+* [Artifacts]
+  * [Using Artifacts to Replicate Builds]
+  * [Using Artifacts to View More Test Information]
+
+*************************************
+Legend for GitHub Actions Build Names
+*************************************
+
+Operating System
+================
+
+* u18/u20 - Ubuntu 18.04/Ubuntu 20.04
+* w16/w19 - Windows 2016/Windows 2019
+* m10 - MacOS 10.15
+
+Build Configuration
+===================
+
+* x86 - Windows 32 bit. If not specified, x64 is implied.
+* re - Release build.  If not specified, Debug is implied.
+* clang5/clang10/gcc6/gcc8/gcc10 - compiler used to build
+OpenDDS. If not specified, the default system compiler is used.
+
+Build Type
+==========
+
+* stat - Static build
+* bsafe/esafe - Base Safety/Extended Safety build
+* sec - Security build
+* asan - Address Sanitizer build
+
+Build Options
+=============
+
+* o1 - enables --optimize
+* d0 - enables --no-debug
+* i0 - enables --no-inline
+* p1 - enables ipv6x
+* w1 - enables wide characters
+* v1 - enables versioned namespace
+* cpp03 - --std=c++03
+* j/j8/j12 - Default System Java/Java8/Java12
+* ace7 - uses ace7tao3 rather than ace6tao2
+* xer0 - disables xerces
+* qt - enables --qt
+* ws - enables --wireshark
+* js0 - enables --no-rapidjson
+
+Feature Mask
+============
+
+This is a mask in an attempt to keep names shorter.
+* FM-08
+  * --no-built-in-topics
+  * --no-content-subscription
+  * --no-ownership-profile
+  * --no-object-model-profile
+  * --no-persistence-profile
+* FM-1f
+  * --no-built-in-topics
+* FM-2c
+  * --no-content-subscription
+  * --no-object-model-profile
+  * --no-persistence-profile
+* FM-2f
+  * --no-content-subscription
+* FM-37
+  * --no-content-filtered-topics
+
+***************************
+build_and_test.yml Workflow
+***************************
+
+Our main workflow which dictates our GitHub Actions run is
+`.github/workflows/build_and_test.yml`. It defines jobs, which are the tasks that
+are run by the CI.
+
+Triggering the Build And Test Workflow
+======================================
+
+There are a couple ways in which a run of build and test workflow can be `started <https://docs.github.com/en/actions/reference/events-that-trigger-workflows>`.
+
+Any pull request targeting master will automatically run the
+OpenDDS workflows. This form of workflow run with simulate a merge
+between the branch and master.
+
+Push events on branches prefixed "gh_wf_" will trigger workflow runs
```

Comment:
```suggestion
Push events on branches prefixed ``gh_wf_`` will trigger workflow runs
```

A trailing underscore is used for some kinds of links. So it has to be formatted or escaped.